src/lib.rs

```diff
@@ -42,19 +49,20 @@ pub fn run(args: Args) {
             .capture_index_for_name(capture_name)
             .expect(&format!("Unknown capture name: `{}`", capture_name))
     });
+
     enumerate_project_files(&*supported_language)
         .par_iter()
-        .flat_map(|project_file_dir_entry| {
-            get_results(
-                &query,
-                project_file_dir_entry.path(),
-                capture_index,
-                language,
-            )
+        .for_each(|project_file_dir_entry| {
+            let mut printer = grep::printer::Standard::new_no_color(io::stdout());
+            let path = project_file_dir_entry.path();
+            let matches = get_results(&query, path, capture_index, language);
+
+            let matcher = TreeSitterMatcher::new(matches);
```

Comment:
We need something to conform to the `Matcher` trait. This seems stupid because it's a trait about incrementally searching through the bytes of a string, but this is really just faking it by getting all results from the file and then feeding them out to the searching via the `Matcher` interface.